nixos/modules/services/web-apps/convos.nix

```diff
@@ -0,0 +1,95 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.convos;
+in
+{
+  options.services.convos = {
+    enable = mkEnableOption "Convos";
+
+    package = mkOption {
+      default = pkgs.convos;
+      defaultText = "pkgs.convos";
+      type = types.package;
+      description = ''
+        Convos package to use.
+      '';
+    };
+
+    listenPort = mkOption {
+      type = types.int;
+      default = 3000;
+      example = 8080;
+      description = ''
+        The port convos should listen on
+      '';
+    };
+
+    listenAddress = mkOption {
+      type = types.str;
+      default = "*";
+      example = "127.0.0.1";
+      description = ''
+        The address convos should listen on
+      '';
+    };
+
+    reverseProxy = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        In case when running behind a reverse proxy, controls whether headers
+        like <literal>X-Forwarded-For</literal> and <literal>X-Forwarded-Proto</literal>
```

Comment:
`X-Forwarded-Proto` is not required. `X-Request-Base` is more important.

Reply:
Thx, updated.